src/Compilers/CSharp/Test/Semantic/Utilities/ValueSetTests.cs

```diff
@@ -0,0 +1,663 @@
+            var none = lt.Complement().Intersect(gt.Complement()).Intersect(eq.Complement());
+            Assert.Equal("NaN", none.ToString());
+            Assert.False(none.IsEmpty);
+        }
+
+        [Fact]
+        public void TestFloat_01()
+        {
+            Assert.Equal("-Inf", ForFloat.Related(LessThan, float.MinValue).ToString());
+            var lt = ForFloat.Related(LessThan, 0.0f);
+            Assert.Equal(FormattableString.Invariant($"-Inf,[{float.MinValue:G9}..{-float.Epsilon:G9}]"), lt.ToString());
+            var gt = ForFloat.Related(GreaterThan, 0.0f);
+            Assert.Equal(FormattableString.Invariant($"Inf,[{float.Epsilon:G9}..{float.MaxValue:G9}]"), gt.ToString());
+            var eq = ForFloat.Related(Equal, 0.0f);
+            Assert.Equal("[0..0]", eq.ToString());
+            var none = lt.Complement().Intersect(gt.Complement()).Intersect(eq.Complement());
+            Assert.Equal("NaN", none.ToString());
+            Assert.False(none.IsEmpty);
+        }
+
+        [Fact]
+        public void TestDouble_03()
+        {
+            Assert.Equal("NaN", ForDouble.Related(Equal, double.NaN).ToString());
+            Assert.Equal("NaN", ForFloat.Related(Equal, float.NaN).ToString());
+            Assert.True(ForDouble.Related(Equal, double.NaN).Any(Equal, double.NaN));
+            Assert.True(ForFloat.Related(Equal, float.NaN).Any(Equal, float.NaN));
+            Assert.Equal("Inf", ForDouble.Related(Equal, double.PositiveInfinity).ToString());
+            Assert.Equal("Inf", ForFloat.Related(Equal, float.PositiveInfinity).ToString());
+            Assert.Equal("-Inf", ForDouble.Related(Equal, double.NegativeInfinity).ToString());
+            Assert.Equal("-Inf", ForFloat.Related(Equal, float.NegativeInfinity).ToString());
+        }
+
+        [Fact]
+        public void TestDouble_04()
+        {
+            var neg = ForDouble.Related(LessThan, 0.0);
+            Assert.True(neg.Any(LessThan, double.MinValue));
+            Assert.False(neg.Any(GreaterThan, double.MaxValue));
+
+            var mi = ForDouble.Related(Equal, double.NegativeInfinity);
+            Assert.True(mi.All(LessThan, 0.0));
+        }
+
+        [Fact]
+        public void TestString_01()
+        {
+            var notaset = ForString.Related(Equal, "a").Complement();
+            var bset = ForString.Related(Equal, "b");
+            var intersect = bset.Intersect(notaset);
+            Assert.False(intersect.Any(Equal, "c"));
+        }
+
+        [Fact]
+        public void TestBool_Cov_01()
+        {
+            var t = ForBool.Related(Equal, true);
+            var f = ForBool.Related(Equal, false);
+            var em = t.Intersect(f);
+            Assert.True(em.IsEmpty);
+            var q = t.Intersect(t);
+            Assert.Same(t, q);
+            IValueSet b = t;
+            Assert.Same(b.Intersect(b), b);
+            Assert.Same(b.Union(b), b);
+            IValueSetFactory bf = ForBool;
```

Comment:
>IValueSetFactory bf = ForBool; [](start = 12, length = 30)

Not used.

Reply:
Will address in next PR.

---
In reply to: [383439717](https://github.com/dotnet/roslyn/pull/41503#discussion_r383439717) [](ancestors = 383439717)